### PR TITLE
ARPBE-73: Making SDK work with Java 17

### PR DESCRIPTION
**Pull request by scriptom**
Overview of changes:
- Update Groovy from v2 to v3
- Move Groovy files away from java folder and into their own folder to be more compliant with maven project structures
- Fixed Test running with both Spock and JUnit4 test runners
- Adding now necessary dependencies in POM, as well as updating several existing dependencies in order to work well with newer versions of Java
- Fixed Travis support for JDK 17

### CODE REVIEW

**Comment by taswartz** [comment]
This looks good to me! I don't have any recommendations here

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,20 @@
 language: java
 
-install: mvn install -DskipTests=true -Dgpg.skip=true
+before_install:
+  - nvm install $NODE_VERSION
+  - curl -s "https://get.sdkman.io" | bash
+  - source "$HOME/.sdkman/bin/sdkman-init.sh"
+  - sdk install java 17.0.5-amzn
+  - sdk use java 17.0.5-amzn
```
Comment (taswartz) [inline .travis.yml:8]: Any chance you could provide a short explanation of why you had to make these changes? Why do we have to use sdkman now?

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+install:
+  - mvn install -DskipTests=true -Dgpg.skip=true
+  - npm install
 
 dist: trusty
 
-jdk:
-  - oraclejdk8
+#jdk:
+#  - oraclejdk8
```
Comment (taswartz) [inline .travis.yml:17]: Why don't we have to tell travis to use oraclejdk17?

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,20 @@
 language: java
 
-install: mvn install -DskipTests=true -Dgpg.skip=true
+before_install:
+  - nvm install $NODE_VERSION
+  - curl -s "https://get.sdkman.io" | bash
+  - source "$HOME/.sdkman/bin/sdkman-init.sh"
+  - sdk install java 17.0.5-amzn
```
Comment (taswartz) [inline .travis.yml:7]: Are we able to use Oracle's Java 17?

File: .travis.yml (hunk 1 above)
Comment (scriptom) [reply .travis.yml]: For completeness: Travis does not support a java 17 image yet, so we need to get jdk 17 ourselves.

File: .travis.yml (hunk 3 above)
Comment (scriptom) [reply .travis.yml]: Changed in [90aae3c](https://github.com/bullhorn/sdk-rest/pull/405/commits/90aae3c585b4c7c1d9f5697a120602c51d4c5e7b). We'll see how the Travis build goes.

File: .travis.yml (hunk 2 above)
Comment (scriptom) [reply .travis.yml:17]: It doesn't exist

**Pull request merged**